docs: Dedicated guide for configuring the Helm chart (#1277)
redpanda-data/documentation: docs/manage/kubernetes/configure-helm-chart.mdx

---
title: Customize the Helm Chart
tags:
  - Kubernetes
---

<head>
    <meta name="title" content="Customize the Helm Chart | Redpanda Docs"/>
    <meta name="description" content="You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys."/>
</head>

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import ValuesYaml from './shared/_values-yaml.mdx'

You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys.

To customize the Redpanda Helm chart, you can override the default values in your own YAML file with the `--values` option or in the command line with the `--set` option.

Helm does a three-way merge with the following:

- Your overrides
- The values in the existing release
- The default values in the new release (if you're upgrading with `helm upgrade`)

To see what options you can override in the chart, use the `helm show values` command:

```bash
helm repo add redpanda https://charts.redpanda.com
helm repo update

> helm show values redpanda/redpanda

Reviewer: Should we mention visual way/browser view of our values: https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values

Author: The `<ValuesYaml>` component below this code injects this link.

If you're upgrading and you already have Redpanda Console installed, set `console.enabled` to `false` to stop Helm from trying to deploy it again.

</TabItem>
<TabItem value="set" label="--set" default>

The `--set` option allows you to specify configuration overrides in the command line. 
For example, you might override the `storage.persistentVolume.storageClass` configuration like so:

```bash
helm upgrade --install redpanda redpanda/redpanda \
    --namespace redpanda --create-namespace \
    --set storage.persistentVolume.storageClass=my-storage-class
```

For more details, see the [Helm documentation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).

The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.

</TabItem>
</Tabs>

## Configure Redpanda Console

> You can configure Redpanda Console in the `console.config` object of the Redpanda Helm chart using the usual [configuration values](../../../reference/console/config).

Reviewer: I wonder if `[configuration values](../../../reference/console/config)` is enough to mention here. Or we could add 
```console
helm repo add redpanda https://charts.redpanda.com
helm repo update
helm show values redpanda/console
```

Just to mention it, but encourage users to use the integration between redpanda and console chart.

Reviewer: Should we mention visual way/browser view of our values: https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values

For example, to enable the admin API for Repanda Console:

<Tabs>
<TabItem value="values" label="--values" default>

```yaml title="console-enable-admin-api.yaml"
console:
  enabled: true
  config: 
    console:
      redpanda:
        adminApi:
          enabled: true
          urls:

>           - http://redpanda-0.redpanda.redpanda.svc.cluster.local.:9644

Reviewer: This is good example, but @joejulian shouldn't this be default for console integration in redpanda chart?

Author: Joe said that he's going to make this the default. I can update this example when it's done.

tags:
  - Kubernetes
---

<head>
    <meta name="title" content="Customize the Helm Chart | Redpanda Docs"/>
    <meta name="description" content="You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys."/>
</head>

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import ValuesYaml from './shared/_values-yaml.mdx'

You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys.

To customize the Redpanda Helm chart, you can override the default values in your own YAML file with the `--values` option or in the command line with the `--set` option.

Helm does a three-way merge with the following:

- Your overrides
- The values in the existing release
- The default values in the new release (if you're upgrading with `helm upgrade`)

To see what options you can override in the chart, use the `helm show values` command:

```bash
helm repo add redpanda https://charts.redpanda.com
helm repo update
helm show values redpanda/redpanda
```

:::tip

<ValuesYaml/>

:::

<Tabs>
<TabItem value="values" label="--values" default>

> The `--values` option allows you to keep your overrides in one or more YAML files. 
> If you provide multiple files to this option and you override the same values in two or more of them, the rightmost file takes precedence.
> For example, you might override the `storage.persistentVolume.storageClass` configuration in a file called `storage-class.yaml`:

Reviewer 2: ```suggestion
The `--values` option enables you to keep your overrides in one or more YAML files. 
If you specify multiple files to this option and then override the same values in two or more of them, the rightmost file takes precedence.
For example, you might override the `storage.persistentVolume.storageClass` configuration in a file called `storage-class.yaml`:
```

```yaml title="storage-class.yaml"
storage:
  persistentVolume:
    storageClass: "my-storage-class"
```

The `helm` command to apply this configuration override looks something like the following:

```bash
helm upgrade --install redpanda redpanda/redpanda \
    --namespace redpanda --create-namespace \
    --values storage-class.yaml --reuse-values
```

> The values in your YAML files override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.

Reviewer 2: ```suggestion
The values in your YAML files override their counterparts in the Helm chart’s `values.yaml` file. Any values not overridden maintain their defaults.
```

Use the `--reuse-values` flag to apply your overrides on top of existing overrides that you've already made. Don't include this flag if you're upgrading to a new version of the Helm chart. If you're upgrading to a new version of the Helm chart, this flag prevents any values in the new release from being applied.

If you're upgrading and you already have Redpanda Console installed, set `console.enabled` to `false` to stop Helm from trying to deploy it again.

</TabItem>
<TabItem value="set" label="--set" default>

> The `--set` option allows you to specify configuration overrides in the command line. 

Reviewer 2: ```suggestion
The `--set` option enables you to specify configuration overrides in the command line. 
```

For example, you might override the `storage.persistentVolume.storageClass` configuration like so:

```bash
helm upgrade --install redpanda redpanda/redpanda \
    --namespace redpanda --create-namespace \
    --set storage.persistentVolume.storageClass=my-storage-class
```

For more details, see the [Helm documentation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).

> The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.

Reviewer 2: ```suggestion
The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values not overridden maintain their defaults.
```

```yaml title="console-enable-admin-api.yaml"
console:
  enabled: true
  config: 
    console:
      redpanda:
        adminApi:
          enabled: true
          urls:
          - http://redpanda-0.redpanda.redpanda.svc.cluster.local.:9644
```

```bash
helm upgrade --install redpanda redpanda/redpanda \
    --namespace redpanda --create-namespace \
    --values console-enable-admin-api.yaml --reuse-values
```

</TabItem>
<TabItem value="set" label="--set" default>

```bash
helm upgrade --install redpanda redpanda/redpanda \
    --namespace redpanda --create-namespace \
    --set console.console.config.redpanda.adminApi.enabled=true \
    --set console.console.config.redpanda.adminApi.urls={"http://redpanda-0.redpanda.redpanda.svc.cluster.local.:9644"}
```

</TabItem>
</Tabs>

If you want to use the separate Redpanda Console Helm chart, disable Redpanda Console in the Redpanda Helm chart with `console.enabled=false`.
To see what options you can override in the Redpanda Console chart, use the `helm show values` command:

```bash
helm repo add redpanda https://charts.redpanda.com
helm repo update
helm show values redpanda/console
```

> For default values and documentation for configuration options, see the [\`values.yaml\`](https://artifacthub.io/packages/helm/redpanda-data/console?modal=values) file.

Reviewer 2: Are you intending for the format tags to appear around values.yaml here? Just checking.

> For default values and documentation for configuration options, see the [\`values.yaml\`](https://artifacthub.io/packages/helm/redpanda-data/console?modal=values) file.

Author: ```suggestion
For default values and documentation for configuration options, see the [`values.yaml`](https://artifacthub.io/packages/helm/redpanda-data/console?modal=values) file.
```
